utils/remote-ext-tests/replay-block/README.md

```diff
@@ -0,0 +1,159 @@
+Regardless of these small inconveniences the `try-runtime` is a good tool whose functionality goes way beyond the scope of the task in question.
+
+In order to mitigate the downsides of the `try-runtime` CLI a dedicated tool has been created. It is dedicated exclusively for remote block execution on top of the on-chain state (currently, only Vara chain is supported).
+
+Advantages with respect to the Substrate' `try-runtime` are:
+- doesn't require enabling additional features;
+- works with the native Runtime (in case the `RuntimeVersion` is as the one of the on-chain Runtime) rendering debugging possible.
+
+There are also some shortcomings though (which can easliy be fixed in later versions), like the fact it can only download state/block from the live chain and not use a local snapshot.
+
+
+<br/>
+
+## Substrate's `try-runtime` CLI command
+
+Substrate provides a rather complex machinery for testing various runtime aspects against real live data (via so-called `remote-externalities`).
+The funtionality covers blocks execution, rening `on_runtime_upgrade` hooks to test storage migration consistency etc.
+It can run against a live chain as well as a downloaded data snapshot.
+Runs locally as a node CLI command.
+
+The inconvenience (even, a downside) is that in order to enable the CLI command and the respective runtime api the node and the runtime have to be built with the `try-runtime` feature. It means that even though the actual "live" runtime is downloaded via RPC from an archive node it can't be used for applying extrinsics in a block (because a real runtime would usually not have been built with the `try-runtime` feature on). So in order to use the `try-runtime` command in CLI we need to prepare a runtime of the same version locally (hoping it does, indeed, correspond to the one currently running on-chain).
+
+Other than that it's a good tool.
+
+<br/>
+
+### Usage examples
+
+* Execute block on Vara live chain
+
+    * current latest finalized block on Vara chain
+
+        ```bash
+        gear try-runtime --chain=vara --runtime vara_runtime.compact.compressed.wasm execute-block live --uri wss://archive-rpc.vara-network.io:443
+        ```
+
+    * at block `$HASH`
+
+        ```bash
+        export HASH=0x8dc1e32576c1ad4e28dc141769576efdbc19d0170d427b69edb2261cfc36e905
+
+        gear try-runtime --chain=vara --runtime vara_runtime.compact.compressed.wasm execute-block live --uri wss://archive-rpc.vara-network.io:443 --at "$HASH"
```

Comment:
can't we use block number here?

Reply:
this refers to the original `try-runtime` CLI. In our tool we do support both the hash and block number